recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -20,10 +20,104 @@ processMatch: []
 preInstall:
   requireAtDiscovery: |
       code=$(curl -I http://169.254.169.254/latest/meta-data/instance-id -w %{response_code} -so '/dev/null')
-      if [ $code == "200" ]; then
-        exit 132
+
+      if [ $code != "200" ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/credentials ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/config ]; then
+        exit 1
+      fi
+
+      if ! command -v aws &> /dev/null; then
+        exit 1
+      fi
+
+      callerIdentity=$(aws sts get-caller-identity 2>&1)
+
+      if [ "$(echo "$callerIdentity" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+        exit 1
+      fi
+
+      function isUserAdmin {
+        username=$(echo "$1" | sed -ne 's/.*"Arn": ".*\/\(.*\)".*/\1/p')
+
+        attachedUserPolicies=$(aws iam list-attached-user-policies --user-name $username 2>&1)
+
+        if [ "$(echo "$attachedUserPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+          false
```

Comment:
@noahmmcgivern this is quite difficult to read and understand what the function isUserAdmin is doing. Could we just return a 1 to indicate the uer is an admin, and a 0 when it's not admin?